keel-core/src/main/kotlin/com/netflix/spinnaker/keel/persistence/ArtifactRepository.kt

```diff
@@ -280,12 +279,12 @@ class NoSuchArtifactException(name: String, type: ArtifactType) :
   constructor(artifact: DeliveryArtifact) : this(artifact.name, artifact.type)
 }
 
-class ArtifactReferenceNotFoundException(deliveryConfig: String, reference: String, type: ArtifactType) :
-  NoSuchEntityException("No $type artifact with reference $reference in delivery config $deliveryConfig is registered")
+class ArtifactReferenceNotFoundException(deliveryConfig: String, reference: String) :
+  NoSuchEntityException("No artifact with reference $reference in delivery config $deliveryConfig is registered")
 
-class ArtifactNotFoundException(name: String, type: ArtifactType, reference: String, deliveryConfig: String?) :
-  NoSuchEntityException("No $type artifact named $name with reference $reference in delivery config $deliveryConfig is registered") {
-  constructor(artifact: DeliveryArtifact) : this(artifact.name, artifact.type, artifact.reference, artifact.deliveryConfigName)
+class ArtifactNotFoundException(reference: String, deliveryConfig: String?) :
```

Comment:
It's confusing that these take the same arguments now, but in different orders, especially since they're both strings. Consider reordering for consistency.

Reply:
Oh wow that is confusing, didn't realize I did this! I'll fix!